### DISCUSSION TITLE
Create peer and channel with sdk

### DISCUSSION THREAD

**Discussion by douglasjava** [hyperledger/fabric-gateway#594] [Q&A] [answered]
Morning, In this newer version of the SDK with java is it possible to create peers and manage the entire network via SDK?

`		<dependency>
			<groupId>org.hyperledger.fabric</groupId>
			<artifactId>fabric-gateway</artifactId>
			<version>1.2.2</version>
		</dependency>
`

`
		<dependency>
			<groupId>org.hyperledger.fabric</groupId>
			<artifactId>fabric-gateway-java</artifactId>
			<version>2.2.8</version>
		</dependency>
`

**Comment by bestbeforetoday** [comment, marked as answer]
Since Fabric v2.2 the focus of the client application SDKs has been on support application runtime (transaction invocation and eventing), not on administrative operations. Neither [fabric-gateway](https://github.com/hyperledger/fabric-gateway) (for Fabric v2.4 and later) or [fabric-gateway-java](https://github.com/hyperledger/fabric-gateway-java) (for earlier Fabric versions) provide any API for creating peers or managing the network. Instead, the Fabric CLI commands have been the primary tool for administrative tasks.

More recently there has been work on [fabric-admin-sdk](https://github.com/hyperledger/fabric-admin-sdk) to provide support for programmatic network configuration. There is so far only an implementation in Go and JavaScript, with no Java implementation. The SDK is also still in development and, while it is pretty much fully functional, is only just reaching API stability.

If you are deploying to Kubernetes then alternatives you could consider are a Kubernetes operator for managing Fabric deployments, such as:

- [hyperledger-labs/fabric-operator](https://github.com/hyperledger-labs/fabric-operator); or
- [hyperledger/bevel-operator-fabric](https://github.com/hyperledger/bevel-operator-fabric)

For a GUI-based experience, there is also [hyperledger-labs/fabric-operations-console](https://github.com/hyperledger-labs/fabric-operations-console).

**Reply by douglasjava** [reply]
thanks!
